lib/core/src/Cardano/Pool/Rank.hs

```diff
@@ -0,0 +1,379 @@
+which results in zero rewards paid out at the beginning of /e5/.
+
+To summarize, in order to make an informed delegation choice
+during epoch /e1/, the delegator needs to know 'RewardInfoPool' where
+
+* 'stakeRelative', 'ownerStake', and 'ownerStakeRelative' are
+   taken at the time of decision (in epoch /e1/).
+* 'ownerPledge', 'margin', 'cost' are the values of the last pool
+   registration certificate
+   from epoch /e0/ in case of an update,
+   or from epoch /e1/ in case of a newly created pool.
+
+For the 'performanceEstimate', it's best to estimate it from recent pool
+block production using functions provided here.
+
+-}
+
+-- | Summary of stake distribution and stake pools obtained from network.
+data StakePoolsSummary = StakePoolsSummary
+    { rewardParams :: RewardParams
+    , pools :: Map PoolId RewardInfoPool
+    } deriving (Show, Eq)
+
+instance Buildable StakePoolsSummary where
+    build StakePoolsSummary{rewardParams,pools} = blockListF' "" id
+        [ "Global reward parameters: " <> build rewardParams
+        , "Individual pools: " <> mapF (Map.toList pools)
+        ]
+
+{-------------------------------------------------------------------------------
+    Reward formulas
+-------------------------------------------------------------------------------}
+fractionOf :: RealFrac r => r -> Coin -> Coin
+fractionOf r (Coin x) = Coin . floor $ r * fromIntegral x
+
+proportionTo :: Coin -> Coin -> Rational
+proportionTo _        (Coin 0) = 0
+proportionTo (Coin x) (Coin y) = fromIntegral x / fromIntegral y
+
+z0 :: RewardParams -> Rational
+z0 RewardParams{nOpt} = 1 / fromIntegral nOpt
+
+epochsPerYear :: Int
+epochsPerYear = 73
+
+-- | The current (yearly) return per unit of stake.
+currentROS :: RewardParams -> RewardInfoPool -> Coin -> Percentage
+currentROS rp RewardInfoPool{..} x
+    | ownerStake < ownerPledge = clipToPercentage 0
+    | otherwise = clipToPercentage $ (1 + astar)^epochsPerYear - 1
+  where
+    s     = clipToPercentage $ ownerPledge `proportionTo` (totalStake rp)
+    sigma = getPercentage stakeRelative + (x `proportionTo` totalStake rp)
+
+    astar
+        | sigma == 0 = 0
+        | otherwise  = shareAfterFees (1/sigma) cost margin fhat
+            `proportionTo` totalStake rp
+    fhat  = optimalRewards rp s sigma
+
+-- | The (yearly) return per unit of stake
+-- for a pool that has reached saturation.
+saturationROS :: RewardParams -> RewardInfoPool -> Percentage
+saturationROS rp RewardInfoPool{..}
+    | ownerStake < ownerPledge = clipToPercentage 0
+    | otherwise = clipToPercentage $ (1 + bstar)^epochsPerYear - 1
+  where
+    s     = clipToPercentage $ ownerPledge `proportionTo` (totalStake rp)
+    sigma = z0 rp -- saturation, never = 0
+
+    bstar = shareAfterFees (1/sigma) cost margin fhat
+        `proportionTo` totalStake rp
+    fhat  = optimalRewards rp s sigma
+
+-- | Non-Myopic Pool Member Rewards
+-- according to Eq.(3) of Section 5.6.4 in SL-D1.
+nonMyopicMemberReward
```

Comment:
Does this one need review as well?

Reply:
Nope, no need to worry about that one. Will be phased out anyway, and was taken from tested in the (now stale) PR https://github.com/input-output-hk/cardano-wallet/pull/2781